src/shared/components/common/markdown-textarea.tsx

```diff
@@ -374,10 +374,63 @@ export class MarkdownTextArea extends Component<
     autosize.update(textarea);
   }
 
-  handleImageUploadPaste(i: MarkdownTextArea, event: any) {
+  handlePaste(i: MarkdownTextArea, event: ClipboardEvent) {
+    if (!event.clipboardData) return;
+
+    // check clipboard files
     const image = event.clipboardData.files[0];
     if (image) {
       i.handleImageUpload(i, image);
+      return;
+    }
+
+    // check clipboard url
+    const url = i.isValidUrl(event.clipboardData.getData("text"));
+    if (url) {
+      i.handleUrlPaste(url, i, event);
+      return;
+    }
+  }
+
+  handleUrlPaste(url: URL, i: MarkdownTextArea, event: ClipboardEvent) {
+    // query textarea element
+    const textarea = document.getElementById(i.id);
+
+    if (textarea instanceof HTMLTextAreaElement) {
+      const { selectionStart, selectionEnd } = textarea;
+
+      // if no selection, just insert url
+      if (selectionStart === selectionEnd) return;
```

Comment:
How is it doing anything here?

Reply:
The comment explains it. If no text is selected (i.e. the curser is in one position and therefore `selectionStart === selectionEnd`) we don't want to apply formatting

Reply:
Gotcha, it has to do with event.preventDefault() not being called.